lighthouse-core/scripts/i18n/collection-util.js

```diff
@@ -0,0 +1,385 @@
+#!/usr/bin/env node
+/**
+* @license Copyright 2019 Google Inc. All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+*/
+'use strict';
+
+/**
+ * @typedef ICUMessageDefn
+ * @property {string} message
+ * @property {string} [description]
+ * @property {Record<string, ICUPlaceholderDefn>} [placeholders]
+ */
+
+/**
+ * @typedef ICUPlaceholderDefn
+ * @property {string} content
+ * @property {string} [example]
+ */
+
+// @ts-ignore - @types/esprima lacks all of these
+function computeDescription(ast, property, value, startRange) {
```

Comment:
seems like most of these aren't really "utils" but just used in one place each. Can we divide up the ones that are used in only one place and have them in place where used? We can do the `if (require.main === module)` trick if the main purpose was to leave the other files executable but have these functions available for testing

Reply:
Done.